### PR TITLE
bsunanda:Phase2-ft09 Add Reco geometry for Fast Timing

### PR DESCRIPTION
**Pull request by bsunanda**
This is the first attempt to provide reconstruction geometry for fast timing layers

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@ianna It will be nice if this could be a part of 8_1_0

**Comment by bsunanda** [comment]
@ianna could you approve this so that it can be a part of 8_1_0?

**Comment by bsunanda** [comment]
@lgray This is the very first attempt - I have not yet tested it. The test function is to be written for this and then I update the scenario

**Comment by lgray** [comment]
@bendavid

**Comment by lgray** [comment]
@bsunanda Thanks, I'll take a look ASAP.

**Comment by bsunanda** [comment]
@ianna Please approve this

**Comment by ianna** [comment]
+1

File: Geometry/CaloTopology/interface/FastTimeTopology.h [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+  /// move the Topology north (increment iy)  
+  virtual DetId  goNorth(const DetId& id) const {
+    return offsetBy(id,0,+1);
+  }
+  virtual std::vector<DetId> north(const DetId& id) const { 
+    DetId nextId= goNorth(id);
+    std::vector<DetId> vNeighborsDetId;
+    if (! (nextId==DetId(0)))
+      vNeighborsDetId.push_back(DetId(nextId.rawId()));
```
Comment (davidlange6) [inline Geometry/CaloTopology/interface/FastTimeTopology.h]: @bsunanda - why the conversion from DetId to uint back to DetId?

Comment (davidlange6) [inline Geometry/CaloTopology/interface/FastTimeTopology.h]: so how do you propose to keep track of the issue (seems more work to remember than to fix..)

**Comment by bsunanda** [comment]
@davidlange6 If you integrate this now I shall put up the next PR once I have the new IB. It will repair not only this Topology class and all the others in this package

**Comment by davidlange6** [comment]
+1
Thanks @bsunanda

**Pull request merged**